File: frontend-angular/src/app/dto/ruta/ruta-dto.ts
import {HorarioDTO} from "./horario-dto";
import {EstacionesDTO} from "./estaciones-dto";
import {BusDTO} from "./bus-dto";
import {ConductorDTO} from "./conductor-dto";

export class RutaDto {
  constructor(
    public codigo: string,
    public horario: HorarioDTO,
    public estaciones: EstacionesDTO[],
    public buses: BusDTO[],
    public conductores: ConductorDTO[]
  ) {}
}

File: backend-springboot/src/main/java/com/javeriana/pontimovil/ponti_movil/dto/gestion_rutas/ruta/RutaDTO.java
package com.javeriana.pontimovil.ponti_movil.dto.gestion_rutas.ruta;

import lombok.Getter;
import lombok.Setter;
import java.io.Serializable;
import lombok.*;
import java.util.List;
import java.util.UUID;

@Getter
@Setter
@AllArgsConstructor
@EqualsAndHashCode
@NoArgsConstructor
public class RutaDTO implements Serializable {
    UUID id;
    String codigo;
    HorarioDTO horario;
    List<EstacionDTO> estaciones;
    List<BusDTO> buses;
    List<ConductorDTO> conductores;
}